torch/_inductor/kernel/mm_common.py

```diff
@@ -146,288 +136,62 @@ def filtered_configs(
-            [16, 32, 64, 128, 256], repeat=3
-        )
-        for num_stages in [1, 2, 3, 4, 5]
-        for num_warps in [2, 4, 8]
-    ]
-)
+if inductor_config.max_autotune_custom_heuristic is None:
+    mm_heuristics = V.choices.config_heuristics
+else:
+    mm_heuristics = inductor_config.max_autotune_custom_heuristic
 
-# these are only used in tuned_mm when AutoHeuristic is enabled
-# the idea is that when AutoHeuristic collects data to learn a heuristic, more configs are autotuned
-# when the learned heuristic is used, the learned heuristic reduces the number of configs down to 10
-# which saves compilation time (since less configs are autotuned) and potentially increase performance
-# because the learned heuristic might predict a config that is not part mm_configs
-extra_mm_kernel_configs = [
-    {"config": (16, 32, 16, 3, 2), "cond": True},
-    {"config": (16, 32, 32, 4, 2), "cond": True},
-    {"config": (16, 32, 32, 5, 2), "cond": True},
-    {"config": (64, 64, 128, 3, 4), "cond": True},
-    {"config": (128, 64, 32, 2, 2), "cond": True},
-    {"config": (128, 64, 64, 3, 8), "cond": True},
-    {"config": (128, 64, 128, 4, 8), "cond": True},
-    {"config": (128, 128, 32, 4, 4), "cond": True},
-    {"config": (128, 128, 64, 3, 8), "cond": True},
-    {"config": (128, 128, 64, 5, 4), "cond": True},
-]
-
-int8_mm_kernel_configs = [
-    {"config": (64, 64, 32, 2, 4), "cond": True},
-    {"config": (64, 128, 32, 3, 4), "cond": True},
-    {"config": (128, 64, 32, 3, 4), "cond": True},
-    {"config": (64, 128, 32, 4, 8), "cond": True},
-    {"config": (128, 64, 32, 4, 8), "cond": True},
-    {"config": (64, 32, 32, 5, 8), "cond": True},
-    {"config": (32, 64, 32, 5, 8), "cond": True},
-    {"config": (128, 128, 32, 2, 8), "cond": True},
-    {"config": (64, 64, 64, 3, 8), "cond": True},
-    # {"config": (32, 32, 128, 2, 4), "cond": True},
-    # {"config": (64, 64, 16, 2, 4), "cond": True},
-    # {"config": (32, 32, 16, 1, 2), "cond": True},
-    {"config": (128, 256, 128, 3, 8), "cond": True},
-    {"config": (256, 128, 128, 3, 8), "cond": True},
-]
-
-# Mixed precision kernel configs for small sizes of m for mm's like (16, 8192) x (8192, 8192).
-mixed_mm_kernel_configs_small_m = [
-    {"config": (16, 128, 256, 3, 4), "cond": True},
-    {"config": (16, 128, 256, 5, 8), "cond": True},
-]
+if inductor_config.max_autotune_gemm_search_space != "EXHAUSTIVE":
+    mm_kernel_configs = mm_heuristics.get_mm_configs()
+else:
+    mm_kernel_configs = mm_heuristics.get_exhaustive_mm_configs()
+
+extra_mm_kernel_configs = mm_heuristics.get_extra_mm_configs()
+int8_mm_kernel_configs = mm_heuristics.get_int8_mm_configs()
+mixed_mm_kernel_configs_small_m = mm_heuristics.get_mixed_mm_configs()
+persistent_mm_kernel_configs = mm_heuristics.get_persistent_mm_configs()
+scaled_mm_kernel_configs = mm_heuristics.get_scaled_mm_configs()
+scaled_persistent_mm_kernel_configs = mm_heuristics.get_scaled_persistent_mm_configs()
```

Comment:
This code should be moved inside functions so it can get overridden by the user (through the V.choices API).  Evaluating in global scope prevents this.